pkg/awsutils/awsutils.go

```diff
@@ -1218,9 +1278,21 @@ func (cache *EC2InstanceMetadataCache) AllocIPAddresses(eniID string, numIPs int
 	}
 
 	log.Infof("Trying to allocate %d IP addresses on ENI %s", needIPs, eniID)
-	input := &ec2.AssignPrivateIpAddressesInput{
-		NetworkInterfaceId:             aws.String(eniID),
-		SecondaryPrivateIpAddressCount: aws.Int64(int64(needIPs)),
+
+	input := &ec2.AssignPrivateIpAddressesInput{}
+
+	if cache.ipv4PrefixDelegation {
```

Comment:
let's create separate function for allocate IpAddress vs allocate IPPrefix.

Reply:
Same as dealloc, I didn't separate it because of code duplication. Will sync up with you offline.